src/node_lkl.cc

```diff
@@ -150,3 +148,39 @@ NAN_METHOD(syscall) {
 		}
 	}
 }
+
+NAN_METHOD(parseDirent64) {
+	// Args:
+	// * Buffer buffer
+	// * int nread
+	// * Array result
+	// Parses buffer (which should be a lkl_linux_dirent64*) and fills the
+	// result array with filenames it finds.
+	// Filenames are Buffers that need to be decoded by the caller.
+	char* buf = (char*) node::Buffer::Data(info[0]);
+	unsigned int nread = info[1]->Uint32Value();
+	assert(nread <= node::Buffer::Length(info[0]));
```

Comment:
parseDirent64 is not an API that we expose, the only callers of this are readdirSync and listdir from readdir.js . We trust them to pass a buffer filled by syscall.getdents64 and a correct nread. 

Using the buffer length as nread would be wrong because getdents64 will not always fill the buffer entirely.